pytest 6: Traceback in pytest.raises contains repr of py.path.local

The [werkzeug](https://github.com/pallets/werkzeug) tests fail with pytest 6:

```python
    def test_import_string_provides_traceback(tmpdir, monkeypatch):
        monkeypatch.syspath_prepend(str(tmpdir))
        # Couple of packages
        dir_a = tmpdir.mkdir("a")
        dir_b = tmpdir.mkdir("b")
        # Totally packages, I promise
        dir_a.join("__init__.py").write("")
        dir_b.join("__init__.py").write("")
        # 'aa.a' that depends on 'bb.b', which in turn has a broken import
        dir_a.join("aa.py").write("from b import bb")
        dir_b.join("bb.py").write("from os import a_typo")
    
        # Do we get all the useful information in the traceback?
        with pytest.raises(ImportError) as baz_exc:
            utils.import_string("a.aa")
        traceback = "".join(str(line) for line in baz_exc.traceback)
>       assert "bb.py':1" in traceback  # a bit different than typical python tb
E       assert "bb.py':1" in "  File local('/home/florian/tmp/werkzeugtest/werkzeug/tests/test_utils.py'):205 in test_import_string_provides_traceb...l('/tmp/pytest-of-florian/pytest-29/test_import_string_provides_tr0/b/bb.py'):1 in <module>\n  from os import a_typo\n"
```

This is because of 2ee90887b77212e2e8f427ed6db9feab85f06b49 (#7274, "code: remove last usage of py.error") - it removed the `str(...)`, but the format string uses `%r`, so now we get the repr of the `py.path.local` object instead of the repr of a string.

I believe this should continue to use `"%r" % str(self.path)` so the output is the same in all cases.

cc @bluetech @hroncok

--- a/changelog/7534.bugfix.rst
+++ b/changelog/7534.bugfix.rst
@@ -1,0 +1,1 @@
+Restored the previous formatting of ``TracebackEntry.__str__`` which was changed by accident.

--- a/src/_pytest/_code/code.py
+++ b/src/_pytest/_code/code.py
@@ -262,7 +262,12 @@ class TracebackEntry:
             raise
         except BaseException:
             line = "???"
-        return "  File %r:%d in %s\n  %s\n" % (self.path, self.lineno + 1, name, line)
+        return "  File %r:%d in %s\n  %s\n" % (
+            str(self.path),
+            self.lineno + 1,
+            name,
+            line,
+        )
 
     @property
     def name(self) -> str:

--- a/src/_pytest/_code/code.py
+++ b/src/_pytest/_code/code.py
@@ -262,7 +262,10 @@ class TracebackEntry:
             raise
         except BaseException:
             line = "???"
-        return "  File '%s':%d in %s\n  %s\n" % (
+        # This output does not quite match Python's repr for traceback entries,
+        # but changing it to do so would break certain plugins.  See
+        # https://github.com/pytest-dev/pytest/pull/7535/ for details.
+        return "  File %r:%d in %s\n  %s\n" % (
             str(self.path),
             self.lineno + 1,
             name,

--- a/src/_pytest/_code/code.py
+++ b/src/_pytest/_code/code.py
@@ -262,7 +262,7 @@ class TracebackEntry:
             raise
         except BaseException:
             line = "???"
-        return "  File %r:%d in %s\n  %s\n" % (
+        return "  File '%s':%d in %s\n  %s\n" % (
             str(self.path),
             self.lineno + 1,
             name,

--- a/testing/code/test_code.py
+++ b/testing/code/test_code.py
@@ -176,12 +176,9 @@ class TestTracebackEntry:
             assert False
         except AssertionError:
             exci = ExceptionInfo.from_current()
-        entry = exci.traceback[0]
-        s = str(entry)
-        print(repr(s))
-        assert re.match(
-            r"  File '.*test_code.py':\d+ in test_tb_entry_str\n  assert False", s
-        )
+        pattern = r"  File '.*test_code.py':\d+ in test_tb_entry_str\n  assert False"
+        entry = str(exci.traceback[0])
+        assert re.match(pattern, entry)
 
 
 class TestReprFuncArgs:

--- a/testing/code/test_code.py
+++ b/testing/code/test_code.py
@@ -1,3 +1,4 @@
+import re
 import sys
 from types import FrameType
 from unittest import mock
@@ -169,6 +170,18 @@ class TestTracebackEntry:
         assert source is not None
         assert len(source) == 6
         assert "assert False" in source[5]
+
+    def test_tb_entry_str(self):
+        try:
+            assert False
+        except AssertionError:
+            exci = ExceptionInfo.from_current()
+        entry = exci.traceback[0]
+        s = str(entry)
+        print(repr(s))
+        assert re.match(
+            r"  File '.*test_code.py':\d+ in test_tb_entry_str\n  assert False", s
+        )
 
 
 class TestReprFuncArgs:

